nih-sparc/sparc-app
# Fix dataset pages fail gracefully aahzrc

# Description

The purpose of this PR is to allow the dataset page to gracefully fail if one of its services is down or errors. In the past, if one of endpoints failed, such as the request for the imagemap via the biolucida endpoint, the page would fail to load. Now, if it fails at those parts, and not show that tab, but all the rest of the page will work.

In addition, I have added an error page that will no  longer show the output of the error via Nuxt, but will show the custom error message and allow the user to continue with the site:
![image](https://user-images.githubusercontent.com/1493195/97617447-29dd0f80-19f4-11eb-972d-60bf617e07cf.png)

## Type of change


# How Has This Been Tested?

- Go to [a dataset](http://localhost:3000/datasets/32?type=dataset)
- Page should load normally 
- Using the Chrome devtools, hit cmd+shift+p and select "Show Request blocking"
<img width="769" alt="Screen Shot 2020-10-29 at 2 34 12 PM" src="https://user-images.githubusercontent.com/1493195/97617472-32cde100-19f4-11eb-9eea-f4a9b85f8953.png">
- Add a pattern called `imagemap`
<img width="319" alt="image" src="https://user-images.githubusercontent.com/1493195/97617356-0a45e700-19f4-11eb-83ec-3a3acbc53b78.png">
- Go to Find Data
- Open the dataset "Identification of peripheral neural circuits that regulate heart rate using optogenetic and viral vector strategies"
- The dataset  page should load, but it won't have the Gallery tab because that request was blocked, and failed gracefully
- Remove the request blocking `imagemap` pattern


# Checklist:
